### PR TITLE
Fix: clear_all_serialized_system was not removing files

### PR DESCRIPTION
**Pull request by daniel-thom**
1. clear_all_serialized_system wasn't looking at the correct paths and so wasn't deleting anything.
2. System metadata files were not being tracked.
3. The code was not using a stable hash function. Different versions of Julia (among other reasons) would produce different directory names.
4. clear_all_serialized_system should be plural. I kept the old name for now, but would prefer to delete it. Let me know what you prefer.

### CODE REVIEW

File: src/utils/utils.jl [hunk 1]
```diff
@@ -42,18 +36,18 @@ function clear_serialized_system(
...
+function clear_all_serialized_systems(path::String)
+    for path in _get_system_directories(path)
+        rm(path; recursive = true)
     end
-    check_serialized_storage()
-    return
 end
 
+clear_all_serialized_systems() = clear_all_serialized_systems(SERIALIZED_DIR)
+clear_all_serialized_system() = clear_all_serialized_systems()
```
Comment (GabrielKS) [inline src/utils/utils.jl:46]: Agreed that it should be plural

File: src/utils/utils.jl [hunk 2]
```diff
@@ -102,3 +96,12 @@ function serialize_case_parameters(case_args::Dict{Symbol, <:Any})
...
 end
+
+function _get_system_directories(path::String)
+    return (
+        joinpath(path, x) for
+        x in readdir(path) if isdir(joinpath(path, x)) && _is_system_hash_name(x)
+    )
+end
+
+_is_system_hash_name(name::String) = isempty(filter(!isxdigit, name))
```
Comment (GabrielKS) [inline src/utils/utils.jl:107]: To be a bit more specific, you could also check length == 64

**Review by GabrielKS — COMMENTED** [review]
Looks fine to me. I haven't downloaded it.

File: src/utils/utils.jl (hunk 2 above)
Comment (daniel-thom) [inline src/utils/utils.jl:107]: This is now a dumb check. The old hash function produced only numbers. I need to make the new function produce only numbers...or just delete all directories.

Comment (GabrielKS) [reply src/utils/utils.jl:107]: You could put a consistent prefix on the directory name before the hash and check for that

**Review by jd-lara — APPROVED** [review]

**Pull request merged**